### ISSUE TITLE
Equivalent kitti_multiloss settings for fcn

### ISSUE THREAD

**Issue by skinnersplace** [MarvinTeichmann/KittiSeg#146]
Hi newbie question, apologies!

I have a workflow currently running with kitti_multiloss.py as the decoder, which produces the following output:

2018-01-02 14:20:36,506 root INFO Raw Results:
2018-01-02 14:20:36,506 root INFO     [train] MaxF1 (raw)    :  95.8023 
2018-01-02 14:20:36,506 root INFO     [train] BestThresh (raw)    :  48.2353 
2018-01-02 14:20:36,507 root INFO     [train] Average Precision (raw)    :  90.6480 
2018-01-02 14:20:36,507 root INFO     [val] MaxF1 (raw)    :  96.0370 
2018-01-02 14:20:36,507 root INFO     [val] BestThresh (raw)    :  48.6275 
2018-01-02 14:20:36,507 root INFO     [val] Average Precision (raw)    :  90.9239 
2018-01-02 14:20:36,507 root INFO     Speed (msec) (raw)    :  268.7356 
2018-01-02 14:20:36,507 root INFO     Speed (fps) (raw)    :  3.7211 
2018-01-02 14:20:36,508 root INFO Smooth Results:
2018-01-02 14:20:36,509 root INFO     [train] MaxF1 (smooth) :  94.4523 
2018-01-02 14:20:36,509 root INFO     [train] BestThresh (smooth) :  55.0980 
2018-01-02 14:20:36,509 root INFO     [train] Average Precision (smooth) :  90.2423 
2018-01-02 14:20:36,509 root INFO     [val] MaxF1 (smooth) :  94.8248 
2018-01-02 14:20:36,509 root INFO     [val] BestThresh (smooth) :  52.5490 
2018-01-02 14:20:36,509 root INFO     [val] Average Precision (smooth) :  90.5789 
2018-01-02 14:20:36,509 root INFO     Speed (msec) (smooth) :  268.6646 
2018-01-02 14:20:36,510 root INFO     Speed (fps) (smooth) :  3.7221 

I wanted to retrain using the pool5 layer, for speed. As kittli_mulltiloss does not pull its segmentations from the hypes.json file, I have switched to the decoder fcn.py.

When i run the same data set with fcn.py I get :

018-01-02 18:50:45,713 root INFO Raw Results:
2018-01-02 18:50:45,713 root INFO     [train] MaxF1 (raw)    :  6.1611 
2018-01-02 18:50:45,713 root INFO     [train] BestThresh (raw)    :  0.0000 
2018-01-02 18:50:45,713 root INFO     [train] Average Precision (raw)    :  3.1784 
2018-01-02 18:50:45,713 root INFO     [val] MaxF1 (raw)    :  6.2836 
2018-01-02 18:50:45,713 root INFO     [val] BestThresh (raw)    :  0.0000 
2018-01-02 18:50:45,713 root INFO     [val] Average Precision (raw)    :  3.2437 
2018-01-02 18:50:45,713 root INFO     Speed (msec) (raw)    :  273.3912 
2018-01-02 18:50:45,714 root INFO     Speed (fps) (raw)    :  3.6578 
2018-01-02 18:50:45,714 root INFO Smooth Results:
2018-01-02 18:50:45,715 root INFO     [train] MaxF1 (smooth) :  6.1611 
2018-01-02 18:50:45,715 root INFO     [train] BestThresh (smooth) :  0.0000 
2018-01-02 18:50:45,715 root INFO     [train] Average Precision (smooth) :  3.1784 
2018-01-02 18:50:45,716 root INFO     [val] MaxF1 (smooth) :  6.2836 
2018-01-02 18:50:45,716 root INFO     [val] BestThresh (smooth) :  0.0000 
2018-01-02 18:50:45,716 root INFO     [val] Average Precision (smooth) :  3.2437 
2018-01-02 18:50:45,716 root INFO     Speed (msec) (smooth) :  274.9923 
2018-01-02 18:50:45,716 root INFO     Speed (fps) (smooth) :  3.6365 

So I believe the segmentation model implemented in fcn8_vgg.py  is not the same as that implemented when I use  fcn. py 

Can someone explain how I need to set up the convolutional layers to get the same results using fcn.py ? 

I am currenly configured as follows:
  }, 
  "loss": "xentropy", 
  "model": {
    "architecture_file": "encoder\\fcn8_vgg.py", 
    "evaluator_file": "evals\\kitti_eval.py", 
    "input_file": "inputs\\fom_seg_input.py", 
    "objective_file": "decoder\\fcn.py", 
    "optimizer_file": "optimizer\\generic_optimizer.py"
  },
  "arch": {
	"fcn_in": "fc7", 
    "image_size": 50, 
    "num_channels": 3, 
    "num_classes": 2, 
    "weight": [1,2], 
    "whitening": false
  }, 
  ], 
  "solver": {
    "adam_eps": 1e-05, 
    "batch_size": 1, 
    "epsilon": 1e-09, 
    "learning_rate": 1e-05, 
    "learning_rate_step": null, 
    "max_steps": 2000, 
    "opt": "Adam", 
    "threads": 8
  }, 
  "use_fc_wd": true, 
  "wd": 0.0005

Many thanks in advance - Skinner

**Comment by skinnersplace** [comment]
would still like some help understanding how to create a hypens config for fcn.py decoder that would recreate the same results that i get using  fcn8_vgg.py with kittimultiloss.
is it possible to reverse engineer the orignal config?
